octoprint_mrbeam/software_update_information.py

```diff
@@ -98,242 +238,209 @@ def switch_software_channel(plugin, channel):
-    )
-
-
-def _set_info_netconnectd_daemon(plugin, tier, beamos_date):
-    if beamos_date > BEAMOS_LEGACY_DATE:
-        branch = "master"
-        pip_command = "sudo /usr/local/netconnectd/venv/bin/pip"
-    else:
-        branch = "mrbeam2-stable"
-        pip_command = GLOBAL_PIP_COMMAND
-    package_name = "netconnectd"
-    # get_package_description does not search for package version.
-    version = get_version_of_pip_module(package_name, pip_command)
-    # get_package_description does not force "develop" branch.
-    return _get_package_description(
-        module_id="netconnectd-daemon",
-        tier=tier,
-        package_name=package_name,
-        displayName="Netconnectd Daemon",
-        displayVersion=version,
-        repo="netconnectd_mrbeam",
-        branch=branch,
-        branch_default=branch,
-        pip="https://github.com/mrbeam/netconnectd_mrbeam/archive/{target_version}.zip",
-        pip_command=pip_command,
-    )
-
-
-def _set_info_iobeam(plugin, tier, beamos_date):
-    if beamos_date > BEAMOS_LEGACY_DATE:
-        pip_command = "sudo /usr/local/iobeam/venv/bin/pip"
-    else:
-        pip_command = GLOBAL_PIP_COMMAND
-    return _get_package_description_with_version(
-        module_id="iobeam",
-        tier=tier,
-        package_name="iobeam",
-        pip_command=pip_command,
-        displayName="iobeam",
-        type="bitbucket_commit",
-        repo="iobeam",
-        api_user="MrBeamDev",
-        api_password="v2T5pFkmdgDqbFBJAqrt",
-        pip="git+ssh://git@bitbucket.org/mrbeam/iobeam.git@{target_version}",
-    )
-
-
-def _set_info_mrb_hw_info(plugin, tier, beamos_date):
-    if beamos_date > BEAMOS_LEGACY_DATE:
-        pip_command = "sudo /usr/local/iobeam/venv/bin/pip"
-    else:
-        pip_command = GLOBAL_PIP_COMMAND
-    return _get_package_description_with_version(
-        module_id="mrb_hw_info",
-        tier=tier,
-        package_name="mrb-hw-info",
-        pip_command=pip_command,
-        displayName="mrb_hw_info",
-        type="bitbucket_commit",
-        repo="mrb_hw_info",
-        api_user="MrBeamDev",
-        api_password="v2T5pFkmdgDqbFBJAqrt",
-        pip="git+ssh://git@bitbucket.org/mrbeam/mrb_hw_info.git@{target_version}",
-    )
+    # fmt: off
+    sw_update_plugin = plugin._plugin_manager.get_plugin_info("softwareupdate").implementation
+    # fmt: on
+    sw_update_plugin._refresh_configured_checks = True
+    sw_update_plugin._version_cache = dict()
+    sw_update_plugin._version_cache_dirty = True
 
 
 @logExceptions
-def _get_octo_plugin_description(module_id, tier, plugin, **kwargs):
-    """Additionally get the version from plugin manager (doesn't it do that by default??)"""
-    # Commented pluginInfo -> If the module is not installed, then it Should be.
-    pluginInfo = plugin._plugin_manager.get_plugin_info(module_id)
-    if pluginInfo is None:
-        display_version = None
+def _set_info_from_cloud_config(plugin, tier, beamos_date, cloud_config):
```

Comment:
why is this a private method? since we are testing it and I don't why shouldn't it be used from the outside...

Reply:
changed the testing to test the public method `get_update_information`